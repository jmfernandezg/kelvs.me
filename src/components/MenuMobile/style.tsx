import styled from "styled-components";

export const MenuMobileContainer = styled.div<{ isActive: boolean }>`
  display: none;
  width: 100%;
  height: 100vh; /* Use vh as a fallback for browsers that do not support Custom Properties */
  height: calc(var(--vh, 1vh) * 100);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: 0 auto;
  overflow: hidden;
  backdrop-filter: blur(8px);
  background-color: ${props => props.theme.bg.light};
  opacity: ${props => (props.isActive ? 0.9 : 0)};
  z-index: ${props => (props.isActive ? 15 : 0)};
  ${({ theme }) => theme.tablet`
    display:flex;
    flex-direction: column;
    align-items: baseline;
    justify-content: space-evenly;
    div:nth-child(2) {
      display: block !important;
    }
  `}
  .dark-mode & {
    background-color: ${props => props.theme.bg.dark};
    span {
      color: ${props => props.theme.bg.light};
    }
  }
  svg {
    opacity: ${props => (props.isActive ? 1 : 0)};
    transition: 700ms;
  }
  @media (prefers-reduced-motion: no-preference) {
    transition: opacity 500ms;
  }
`;

export const LinkWrapper = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  align-content: center;
  justify-content: center;
  font-size: 28px;
  width: 100%;
  transition: opacity 0.25s ease-in-out;
  span {
    text-align: left;
    display: block;
    color: ${props => props.theme.text.dark};
  }
  a {
    display: inline-block;
    position: relative;
    padding: 15px 0;
    text-decoration: none;
  }
`;

export const HamburgerWrapper = styled.div<{ isActive: boolean }>`
  display: none;
  ${({ theme }) => theme.tablet`display:block;`}
  svg {
    position: relative;
    height: 2em;
    cursor: pointer;
    -webkit-tap-highlight-color: transparent;
    transition: transform 400ms;
    user-select: none;
    z-index: 20;

    .line {
      fill: none;
      transition: stroke-dasharray 400ms, stroke-dashoffset 400ms;
      stroke: currentColor;
      stroke-width: 5.5;
      stroke-linecap: round;
    }
  }

  svg .top {
    stroke-dasharray: 40 172;
  }
  svg .middle {
    stroke-dasharray: 40 111;
  }
  svg .bottom {
    stroke-dasharray: 40 172;
  }

  ${props =>
    props.isActive
      ? `svg .top {
      stroke-dashoffset: -132px;
    }
    svg .middle {
      stroke-dashoffset: -71px;
    }
    svg .bottom {
      stroke-dashoffset: -132px;
    }`
      : ""}
`;
