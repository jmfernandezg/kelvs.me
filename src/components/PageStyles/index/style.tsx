import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  min-height: calc(100vh - 7rem);
  margin-top: 3em;
  ${({ theme }) => theme.desktop`
    justify-content: center;
    margin-bottom: 0;
    margin-top: 0;
    min-height: calc(65vh - 7rem);
    `}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 0 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.bigTablet`
    margin: 2em 0;
    svg {
      height: 1.1em;
    }
  `}
  ${({ theme }) => theme.thone`
    width: 40%;
  `}
`;

export const FeaturedProjects = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  align-items: center;
`;

export const AppBox = styled.div`
  background: #fff 0 0 no-repeat padding-box;
  box-shadow: 0 3px 6px rgba(0, 0, 0, 0.16);
  width: 32%;
  padding: 0.75em;
  margin-bottom: 1em;
  position: relative;
  transition: transform 0.2s;
  min-height: 290px;
  &:hover {
    transform: scale(1.025);
  }
  color: ${props => props.theme.text.dark};
  img {
    border: 1px solid #000;
    width: 100%;
    height: auto;
  }
  span {
    margin-right: 3px;
    background-color: #ff4e4e;
    color: #fff;
    padding: 0 3px 1px 3px;
    line-height: normal;
    font-size: 12px;
    font-weight: 300;
    letter-spacing: 1px;
    border-radius: 3px;
  }
  p {
    margin-bottom: 0;
  }
  ${({ theme }) => theme.tablet`
    width: 48%;
    p {
      font-size: 0.9rem;
    }
  `}
  ${({ theme }) => theme.thone`width: 100%;`}
`;

export const H1 = styled.h1`
  margin: 2.5em 0;
  ${({ theme }) => theme.tablet`
    margin: 2em 0;
    font-size: 1.125rem;
  `}
  font-weight: 700;
`;
