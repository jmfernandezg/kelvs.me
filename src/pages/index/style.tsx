import styled from "styled-components";

export const HeroWrapper = styled.div`
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  min-height: 40vh;
  ${({ theme }) => theme.tablet`justify-content: center;`}
`;

// TODO: put a div in the hero that can be tilted like in react spring
export const HeroOne = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: flex-start;
  font-family: "Caveat", cursive;
  h1 {
    display: inline-block;
    overflow: visible;
    width: 100%;
    flex: 0 auto;
    font-size: 4em;
    font-weight: 400;
    text-align: center;
    ${({ theme }) => theme.thone`font-size: 2em;`}
  }
  span {
    font-weight: 700;
  }
`;

export const HeroTwo = styled.div`
  h1 {
    position: relative;
    font-size: 3.5em;
    font-weight: 400;
    text-align: center;
    line-height: 58px;
    font-family: "Lato", sans-serif;
    ${({ theme }) => theme.thone`font-size: 2em;`}
    ${({ theme }) => theme.phablet`
      font-size: 1.5em;
      line-height: 35px;
    `}
  }
  span {
    padding-right: 5px;
    padding-left: 5px;
    background: linear-gradient(
      transparent 10%,
      #495867 10%,
      #495867 100%,
      transparent 90%
    );
    color: #fff;
    font-weight: 300;
    text-align: justify;
  }
`;

export const CallToAction = styled.div`
  margin: 2em 0 4em 0;
  font-size: 15px;
  ${({ theme }) => theme.tablet`margin: 2em 0 4em 0;`}
  a {
    padding: 0 20px;
    height: 25px;
  }
  a::before,
  a::after {
    position: absolute;
    width: 25px;
    height: 1px;
    background: ${props => props.theme.text.dark};
    content: "";
    opacity: 0.2;
    transition: all 0.3s;
    pointer-events: none;
  }
  .dark-mode & a::before,
  .dark-mode & a::after {
    background: ${props => props.theme.text.light};
  }
  a:hover {
    opacity: 0.9;
  }
  a::before {
    top: 0;
    left: 0;
    transform: rotate(90deg);
    transform-origin: 0 0;
  }
  a::after {
    right: 0;
    bottom: 0;
    transform: rotate(90deg);
    transform-origin: 100% 0;
  }
  a:hover::before,
  a:hover::after,
  a:focus::before,
  a:focus::after {
    opacity: 1;
  }
  a:hover::before,
  a:focus::before {
    left: 50%;
    transform: rotate(0deg) translateX(-50%);
  }

  a:hover::after,
  a:focus::after {
    right: 50%;
    transform: rotate(0deg) translateX(50%);
  }
`;

export const Contact = styled.div`
  width: 20%;
  display: flex;
  justify-content: space-around;
  svg {
    height: 1.5em;
    width: 100%;
    fill: ${props => props.theme.text.dark};
  }
  .dark-mode & svg {
    fill: ${props => props.theme.text.light};
  }
  ${({ theme }) => theme.tablet`
    width: 40%;
    svg {
      height: 1.1em;
    }
  `}
`;
