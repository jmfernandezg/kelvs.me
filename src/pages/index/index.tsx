/* eslint-disable react/jsx-one-expression-per-line */
import Link from "next/link";
import Layout from "~/components/Layout";
import GmailSVG from "~/components/Icons/Gmail";
import GithubSVG from "~/components/Icons/Github";
import LinkedinSVG from "~/components/Icons/LinkedIn";
import { Container, HorizontalBar } from "~/components/PageStyles/common/style";
import {
  HeroWrapper,
  HeroOne,
  HeroTwo,
  CallToAction,
  Contact,
} from "~/components/PageStyles/index/style";

// https://www.justinewargnier.com/
// https://simpleicons.org/
// TODO: animate full stack dev text switch using react springs
export default function Home() {
  return (
    <Layout>
      <Container>
        <HeroWrapper>
          <HeroOne>
            <h1>
              Hello,&nbsp;
              <span>my name is Kelvin.</span>
            </h1>
            <HorizontalBar />
          </HeroOne>
          <HeroTwo>
            <h1>
              I&apos;m a <span>full&#8209;stack developer</span>
              &nbsp;and an open source enthusiast.
            </h1>
          </HeroTwo>
        </HeroWrapper>
        <CallToAction>
          <Link href="/about">
            <a>Learn more about me</a>
          </Link>
        </CallToAction>
        <Contact>
          <a href="https://github.com/kelvinabella">
            <GithubSVG />
          </a>
          <a href="https://www.linkedin.com/in/kelvinabella">
            <LinkedinSVG />
          </a>
          <a href="mailto:abellakelvin.ka@gmail.com">
            <GmailSVG />
          </a>
        </Contact>
      </Container>
    </Layout>
  );
}
