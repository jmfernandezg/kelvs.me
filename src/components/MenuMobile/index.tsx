/* eslint-disable react/no-array-index-key */
import Link from "next/link";
import { useEffect, useState } from "react";
import { animated, useTrail } from "react-spring";
import DarkModeIcon from "../DarkModeIcon";
import Hamburger from "../Icons/Hamburger";
import { useScrollDisabler, mobileHeight } from "~/helpers/index";
import { MenuMobileContainer, LinkWrapper, HamburgerWrapper } from "./style";

export default function MenuMobile() {
  const items = ["About", "Blog", "Projects", "Contact", "AMA"];
  const [isActive, setIsActive] = useState(false);

  const springs = useTrail(items.length, {
    to: { transform: isActive ? "translateX(70px)" : "translateX(-150px)" },
    config: { tension: 250 },
  });

  useScrollDisabler(isActive);

  useEffect(() => {
    mobileHeight();
  }, []);

  return (
    <>
      <MenuMobileContainer isActive={isActive}>
        <LinkWrapper>
          {springs.map((animation, index) => (
            <animated.span style={animation} key={index}>
              <Link href={`/${items[index].toLowerCase()}`}>
                <a>{items[index]}</a>
              </Link>
            </animated.span>
          ))}
        </LinkWrapper>
        <DarkModeIcon />
      </MenuMobileContainer>
      <HamburgerWrapper
        isActive={isActive}
        onClick={() => setIsActive(!isActive)}
      >
        <Hamburger />
      </HamburgerWrapper>
    </>
  );
}
