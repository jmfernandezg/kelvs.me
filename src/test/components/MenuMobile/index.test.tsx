import React from "react";
import userEvent from "@testing-library/user-event";
import { render, screen } from "~/test/utils/index";
import MenuMobile from "~/components/MenuMobile";

// https://medium.com/@rickhanlonii/understanding-jest-mocks-f0046c68e53c
describe("MenuMobile component", () => {
  it("renders MenuMobile correctly", () => {
    const { asFragment } = render(<MenuMobile />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the proper icon depending whether dark mode state value is true or false", () => {
    const { container } = render(<MenuMobile />);
    const hamburger = screen.getByTitle("Hamburger Menu");
    window.scrollTo = jest.fn();
    // svg should be loaded
    userEvent.click(hamburger);
    expect(hamburger).toContainHTML("<title>Hamburger Menu</title>");
    expect(container).toContainHTML(`<a href="/about">About</a>`);
  });
});
