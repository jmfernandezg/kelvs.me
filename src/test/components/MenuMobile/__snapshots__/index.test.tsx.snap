// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuMobile component renders MenuMobile correctly 1`] = `
<DocumentFragment>
  <div
    class="style__MenuMobileContainer-sc-1d9cvo5-0 dWlKSS"
  >
    <div
      class="style__LinkWrapper-sc-1d9cvo5-1 biJFEi"
    >
      <span
        style="transform: translateX(-150px);"
      >
        <a
          href="/about"
        >
          About
        </a>
      </span>
      <span
        style="transform: translateX(-150px);"
      >
        <a
          href="/blog"
        >
          Blog
        </a>
      </span>
      <span
        style="transform: translateX(-150px);"
      >
        <a
          href="/projects"
        >
          Projects
        </a>
      </span>
      <span
        style="transform: translateX(-150px);"
      >
        <a
          href="/contact"
        >
          Contact
        </a>
      </span>
      <span
        style="transform: translateX(-150px);"
      >
        <a
          href="/ama"
        >
          AMA
        </a>
      </span>
    </div>
    <div
      class="style__IconContainer-sc-1kqqcq8-0 fgJKFR"
    >
      <span
        aria-label="dark mode"
        class="style__AnimatedSpan-sc-1kqqcq8-1 dMwwia"
        role="button"
        style="transform: rotate(90deg);"
        tabindex="0"
      >
        <svg
          class="sun"
          height="480pt"
          viewBox="0 0 480 480"
          width="480pt"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="m368 240c0 70.691406-57.308594 128-128 128s-128-57.308594-128-128 57.308594-128 128-128 128 57.308594 128 128zm0 0"
          />
          <path
            d="m248 472v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8s8-3.582031 8-8zm0 0"
          />
          <path
            d="m240 80c4.417969 0 8-3.582031 8-8v-64c0-4.417969-3.582031-8-8-8s-8 3.582031-8 8v64c0 4.417969 3.582031 8 8 8zm0 0"
          />
          <path
            d="m114.34375 125.65625c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844l-48-48c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844zm0 0"
          />
          <path
            d="m365.65625 354.34375c-3.140625-3.03125-8.128906-2.988281-11.214844.097656-3.085937 3.085938-3.128906 8.074219-.097656 11.214844l48 48c3.140625 3.03125 8.128906 2.988281 11.214844-.097656 3.085937-3.085938 3.128906-8.074219.097656-11.214844zm0 0"
          />
          <path
            d="m360 128c2.121094 0 4.15625-.84375 5.65625-2.34375l48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.285156 2.289062-2.972656 5.730469-1.734375 8.71875s4.15625 4.9375 7.390625 4.9375zm0 0"
          />
          <path
            d="m472 232h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8zm0 0"
          />
          <path
            d="m8 248h64c4.417969 0 8-3.582031 8-8s-3.582031-8-8-8h-64c-4.417969 0-8 3.582031-8 8s3.582031 8 8 8zm0 0"
          />
          <path
            d="m77.65625 413.65625 48-48c3.03125-3.140625 2.988281-8.128906-.097656-11.214844-3.085938-3.085937-8.074219-3.128906-11.214844-.097656l-48 48c-2.078125 2.007812-2.914062 4.984375-2.179688 7.78125.730469 2.796875 2.914063 4.980469 5.710938 5.710938 2.796875.734374 5.773438-.101563 7.78125-2.179688zm0 0"
          />
        </svg>
      </span>
    </div>
  </div>
  <div
    class="style__HamburgerWrapper-sc-1d9cvo5-2 ekkuZo"
  >
    <svg
      height="35"
      viewBox="10 10 70 70"
      width="35"
    >
      <title>
        Hamburger Menu
      </title>
      <path
        class="line top"
        d="m 30,33 h 40 c 13.100415,0 14.380204,31.80258 6.899646,33.421777 -24.612039,5.327373 9.016154,-52.337577 -12.75751,-30.563913 l -28.284272,28.284272"
      />
      <path
        class="line middle"
        d="m 70,50 c 0,0 -32.213436,0 -40,0 -7.786564,0 -6.428571,-4.640244 -6.428571,-8.571429 0,-5.895471 6.073743,-11.783399 12.286435,-5.570707 6.212692,6.212692 28.284272,28.284272 28.284272,28.284272"
      />
      <path
        class="line bottom"
        d="m 69.575405,67.073826 h -40 c -13.100415,0 -14.380204,-31.80258 -6.899646,-33.421777 24.612039,-5.327373 -9.016154,52.337577 12.75751,30.563913 l 28.284272,-28.284272"
      />
    </svg>
  </div>
</DocumentFragment>
`;
